### PR TITLE
Benchmark/fe11939/integration

### PR DESCRIPTION
**Pull request by yeudit**
cleanup code
improve query results
handle errors
and few more fixes to code

### CODE REVIEW

**Review by mikekcs — COMMENTED** [review]
Api 409 to lock  code 423
Api - check error
Api test_is_running to 
Curl - remove  console.log("this.configService.get('algorithms',[])")
Curl - change 409
Run k8s - remove service account
Run k8s - remove timeout annotations from portal
Run k8s - remove comment in prometheus.yaml

File: api/app.py [hunk 1]
```diff
@@ -38,19 +34,48 @@ def get_algorithms():
 @app.route('/analyze', methods=['POST'])
 @cross_origin(origin=['*'],supports_credentials=True)
 def analyze():
-    # export platform info to prometheus
-    response = requests.post(qujata_platform_exporter_target + "/export-platform-info")
-
     global test_is_running
```
Comment (litalmason) [inline api/app.py:37]: how come we define this global inside this API? I can also see this var on the top of the page, but not a global var.
Won't
**global** test_is_running = False
on the top of the page work?

File: api/app.py [hunk 2]
```diff
@@ -38,19 +34,48 @@ def get_algorithms():
...
+    error = __validate(data)
+    if error != None:
+        return error
+
+    test_is_running = True
+    start_time = int(datetime.timestamp(datetime.now() - timedelta(seconds=60)) * 1000)
+    error = __start_analyze(data)
+    if error != None:
+        return error
+    end_time = int(datetime.timestamp(datetime.now() + timedelta(seconds=90)) * 1000)
```
Comment (litalmason) [inline api/app.py:50]: what is the reason we're adding **90** seconds specifically? and **60** on the start time?

File: api/app.py [hunk 3]
```diff
@@ -38,19 +34,48 @@ def get_algorithms():
...
+    global test_is_running
     if not data or 'algorithms' not in data:
-        return jsonify({'error': 'Invalid data provided'}), 400
+        return jsonify({'error': 'Invalid data provided', 'message': 'missing algorithms'}), 400
+    if data['iterationsCount'] < 500 or data['iterationsCount'] > 100000:
+        return jsonify({'error': 'Invalid data provided', 'message': 'iterationsCount must be greater then 500 and less then 100000'}), 400
     if test_is_running:
-        return jsonify({'error': 'Analisys test is running now'}), 409
-    test_is_running = True
+        return jsonify({'error': 'Current test is still running', 'message':'The previous test is still running. Please try again in few minutes'}), 423
```
Comment (litalmason) [inline api/app.py:72]: 423 is very accurate for this error, nice! 
consider using "Constants" for all HTTP codes
For example

LOCKED = 423
BAD_REQUEST = 400

**Review by litalmason — CHANGES_REQUESTED** [review]

File: api/app.py [hunk 4]
```diff
@@ -68,24 +93,9 @@ def analyze():
         response = requests.post(qujata_curl_target + "/curl", headers=headers, json=payload, timeout=request_timeout)
         # Print response details
         print('Status code:', response.status_code)
+        if(response.status_code < 200 or response.status_code > 299):
```
Comment (litalmason) [inline api/app.py:96]: I recommend to extract this line
response.status_code < 200 or response.status_code > 299
to a separate function

File: api/README.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+cd qujata/api
+```
+2. Install python packages:
+```bash
+pip3 install -r requirements.txt
+```
+
+3. Running the application:
+```bash
+python3 app.py
```
Comment (mikekcs) [inline api/README.md:26]: This can be run with python 2.x?

**Review by mikekcs — APPROVED** [review]

File: api/app.py [hunk 6]
```diff
@@ -2,28 +2,31 @@
...
+qujata_curl_target = os.environ.get('CURL_URL', "http://localhost:3010")
 request_timeout = os.environ.get('REQUEST_TIMEOUT', 900)
-test_is_running = False
+min_iterations = int(os.environ.get('MIN_ITERATIONS', 500))
+max_iterations = int(os.environ.get('MAX_ITERATIONS', 100000))
+process_is_running = False
 
+# constants
+LOCKED = 423
+BAD_REQUEST = 400
```
Comment (mikekcs) [inline api/app.py:29]: better to name it something like HTTP_X

**Review by litalmason — APPROVED** [review]

**Pull request merged**